include/mnist.h

```diff
@@ -0,0 +1,62 @@
+#include <stdio.h>
+#include <stdlib.h>
+
+#define NUM_IMAGES 1797
+
+int LABELS[NUM_IMAGES];
+float FLAT_IMAGES[NUM_IMAGES * 64];
+float IMAGES[NUM_IMAGES][64];
+
+void load_images()
```

Comment:
Is there a specific reason these implementations are in the header file? You might get linker errors if you #include this in more than one compilation unit. Consider moving these implementations to a separate .c file, or if there's a reason they're in the header, wrap them with this:


```c
#ifndef __MNIST_H__
#define __MNIST_H__
// ...
#endif
```